python/link_prediction.py

```diff
@@ -0,0 +1,553 @@
+        ctx (mgp.ProcCtx): A reference to the context execution. 
+        num_runs (int): Number of runs.
+
+    Returns:
+        mgp.Record:
+            status (str): Status message
+            test_results: Average test results obtained after training for num_runs times. 
+    """
+    # NOTE: THIS SHOULD BE HIDDEN FOR OUT USERS, BUT ENABLE THIS IN THE DEVELOPMENT SUPPORT
+
+    # Get the global context
+    global training_results, validation_results, predictor, model
+    # Reset parameters obtained from the training
+    _reset_train_predict_parameters()
+
+    # Get DGL graph data representation
+    graph, new_to_old, _ = _get_dgl_graph_data(ctx)  # dgl representation of the graph and dict new to old index
+    """ if test_conversion(graph=graph, new_to_old=new_to_old, ctx=ctx, node_id_property=link_prediction_parameters.node_id_property, node_features_property=link_prediction_parameters.node_features_property) is False:
+        print("Remapping failed")
+        return mgp.Record(status="Preprocessing failed", metrics=[])
+    """
+
+    """ Train g is a graph which has removed test edges. Others are positive and negative train and test graphs
+    """
+
+    # Split the data
+    train_g, train_pos_g, train_neg_g, test_pos_g, test_neg_g = preprocess(graph, link_prediction_parameters.split_ratio)
+    if train_g is None or train_pos_g is None or train_neg_g is None or test_pos_g is None or test_neg_g is None:
+        print("Preprocessing failed. ")
+        return mgp.Record(status="Preprocessing failed", metrics=[])
+
+    test_results = benchmark.get_avg_seed_results(num_runs, link_prediction_parameters.hidden_features_size, link_prediction_parameters.layer_type,
+                                                  link_prediction_parameters.num_epochs, link_prediction_parameters.optimizer, link_prediction_parameters.learning_rate,
+                                                  link_prediction_parameters.node_features_property, link_prediction_parameters.console_log_freq, link_prediction_parameters.checkpoint_freq,
+                                                  link_prediction_parameters.aggregator, link_prediction_parameters.metrics, link_prediction_parameters.predictor_type, link_prediction_parameters.hidden_features_size[-1],
+                                                  link_prediction_parameters.attn_num_heads, link_prediction_parameters.tr_acc_patience, train_g, train_pos_g, train_neg_g, test_pos_g, test_neg_g)
+
+    return mgp.Record(status="OK", test_results=test_results)
+
+
+@mgp.read_proc
+def get_training_results(ctx: mgp.ProcCtx) -> mgp.Record(tr_metrics=mgp.Any, val_metrics=mgp.Any):
+    """This method is used when user wants to get performance data obtained from the last training. It is in the form of list of records where each record is a Dict[metric_name, metric_value]. Training and validation
+    results are returned. 
+
+    Args:
+        ctx (mgp.ProcCtx): Reference to the context execution
```

Comment:
On the above line define global training_results, and check what happens if there are none.